doc/design/ib-vf-configuration.md

```diff
@@ -0,0 +1,123 @@
+or the PF GUID and has either a list or a range of VF GUIDs.
+
+1. A script creating the config file is deployed to the host and creates a static GUID config file
+2. SR-IOV network operator reads the file and assign GUIDs when IB VFs are created
+3. Users employ [PKey selector](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pull/517) in sriov-network-device-plugin to create PKey-specific resource pools
+   3.1. Note: When a VF gets assigned with GUID, it communicates with subnet manager and receives the PKey associated with that GUID. This PKey is stored in the sysfs and device-plugin reads it to populate pools.
+4. Users create an IB network CR
+5. Users create pods and request resources from the specific pools
+
+### Implementation Details/Notes/Constraints
+
+There can be fewer VFs created than GUIDs. To persist the dynamic nature of the SR-IOV Network operator,
+it’s proposed not to return an error in this case but assign as many GUIDs as possible.
+To ensure that nothing breaks when users add/remove VFs, the GUID distribution order should always be the same for each individual host.
+
+Parallel NIC configuration should be supported by making the GUID immutable. VF id should be used to determine the correct GUID for each VF.
+
+If there are fewer GUIDs than VFs, then all the GUIDs should be assigned. For the rest of VFs, apply random GUIDs as it's been done before.
+
+If config file doesn't exist in the predefined location, proceed as usual and assign random GUIDs to VFs.
+
+### Config file
+
+Config file can be stored at `/etc/sriov-operator/infiniband/guids`. This location is writable across most cloud platforms.
+
+Example of the config file:
+
+```json
+[
+  {
+    "pci_address": "<pci_address_1>",
+    "guids": [
+      "02:00:00:00:00:00:00:00",
+      "02:00:00:00:00:00:00:01"
+    ]
+  },
+  {
+    "pf_guid": "<pf_guid_2>",
+    "rangeStart": "02:00:00:00:00:aa:00:02",
+    "rangeEnd": "02:00:00:00:00:aa:00:0a"
+  }
+]
+```
+
+Requirements for the config file:
+
+* `pci_adress` and `pf_guid` cannot be set at the same time for a single device - should return an error
+* if the list contains multiple entries for the same device, the first one shall be taken
+* `rangeStart` and `rangeEnd` are both included in the range
+* `guids` list and range cannot be bot set at the same time for a single device - should return an error
+
+### Test Plan
+
+* Unit tests will be implemented for new logic.
+
+## Alternative solution
+
+The alternative solution is also based on the GUID configuration file being deployed on the host.
+The difference here is that GUID assignment is done on the cni level when a VF is allocated to a pod.
+ib-sriov-cni manages a host-local per-PF pool of allocated/free GUIDs and dynamically allocates the next free GUID to an allocated VF.
+
+### Workflow:
+
+1. A script is deployed to the host and creates a static GUID config file
```

Comment:
this is out of the scope of the operator right? if that is the case please mention that and maybe give an example of how the customer is supposed to configure the file

Reply:
Mentioned that the step is out of scope of the operator.